.github/workflows/ci.yaml

```diff
@@ -30,6 +30,7 @@ jobs:
       matrix:
         os:
           - ubuntu-20.04
+          - macos-latest
```

Comment:
It is tricky to enable macos support like this, because we did that once but the macos runners were just too slow. Ideally we should only run it in the merge queue and pushes, as we do with arm64 now, but then it can't be part of the matrix, and would need a separate job I guess. Not 100% sure if there is a way to have `if` rules for specific items in the matrix.